### ISSUE TITLE
add subscriptions support to Graphql

### ISSUE THREAD

**Issue by srghma** [ethereum/go-ethereum#24518]
# Rationale

> Why should this feature exist?

grapqh is easier to debug and try-out than rpc

> What are the use-cases?

1. Subscribe to log changes in `/graphql/ui` (probably arguments should contain fields for `eth_newFilter`)
2. ... what else can be implemented using graphql subscriptions? 

# Implementation

> Do you have ideas regarding the implementation of this feature?

this is how postgraphile adds subscriptions 

https://github.com/graphile/postgraphile/blob/7c15ce5b275ff41969c631c1d6d1d46c0bcdb6dc/postgraphiql/src/components/PostGraphiQL.js#L168

https://www.graphile.org/postgraphile/subscriptions/

> Are you willing to implement this feature?

not sure

**Comment by s1na** [comment]
I agree this is a useful feature and would also make the graphql endpoint more self-contained. Though it is a bigger feature and it'd be nice to have cross-client coordination on its design, which will be easier after the schema becomes a living document as per https://github.com/ethereum/execution-apis/issues/163.

**Comment by srghma** [comment]
will rename since it seems like the filter functionality is the only one that is left to implement

https://eips.ethereum.org/EIPS/eip-1767

![2022-03-09-04pm-38-49-screenshot](https://user-images.githubusercontent.com/7573215/157463507-0062158c-a1b3-464b-98b3-de234643c3a3.png)
